include/zephyr/acpi/acpi.h

```diff
@@ -5,6 +5,7 @@
  */
 #ifndef ZEPHYR_INCLUDE_DRIVERS_ACPI_H_
 #define ZEPHYR_INCLUDE_DRIVERS_ACPI_H_
+#ifdef CONFIG_ACPI
```

Comment:
shouldn't we just prevent the inclusion of this header if !defined CONFIG_ACPI?

Reply:
Yeah that's what I was propsing as well, I think it's a better idea, just in case there's more headers to be added to the conditional.

Reply:
actually https://github.com/zephyrproject-rtos/zephyr/pull/62682 is what I'd expect

Reply:
Sure, let me change that

Reply:
@gmarull @fabiobaltieri done

Reply:
Note that the reason I did it this way is that @tejlmand considered it a better solution :)